### PR TITLE
Do not append tolerance with zero value to diff message

### PR DESCRIPTION
**Pull request by valfirst**

### CODE REVIEW

File: json-unit-core/src/main/java/net/javacrumbs/jsonunit/core/internal/Diff.java [hunk 1]
```diff
@@ -62,12 +63,12 @@
  */
 public class Diff {
 
-    private static final Pattern ANY_NUMBER_PLACEHOLDER = Pattern.compile("[$#]\\{json-unit.any-number\\}");
-    private static final Pattern ANY_BOOLEAN_PLACEHOLDER = Pattern.compile("[$#]\\{json-unit.any-boolean\\}");
-    private static final Pattern ANY_STRING_PLACEHOLDER = Pattern.compile("[$#]\\{json-unit.any-string\\}");
+    private static final Pattern ANY_NUMBER_PLACEHOLDER = Pattern.compile("[$#]\\{json-unit.any-number}");
```
Comment (lukas-krecan) [inline json-unit-core/src/main/java/net/javacrumbs/jsonunit/core/internal/Diff.java:66]: Please do not do this, it breaks Android tests https://github.com/lukas-krecan/JsonUnit/pull/227

**Comment by lukas-krecan** [comment]
Hi, thanks a lot  for the pull request. Why do you want to remove the tolerance from the log? I find it helpful in cases when the user does not know that he is supposed to set the tolerance.

**Comment by valfirst** [comment]
@lukas-krecan in our case `0` tolerance is used to ignore int vs float types (`1` vs `1.0`)  and the information about tolerance looks unexpected for our users (because they don't actually define it). Another solution is to make this behavior (int vs float comparison) configurable. What do you think?

**Comment by lukas-krecan** [comment]
Let me think about it, I have to re-read the code. Thanks again.

**Comment by lukas-krecan** [comment]
Released as 2.17.0

**Comment by valfirst** [comment]
@lukas-krecan thank you!

**Pull request merged**